component.ts

```diff
@@ -0,0 +1,47 @@
+function fetchResults(query: string) {
+  return fetch(`API_DOMAIN/search?q=${query}`);
+}
+
+function Phonebook() {
+  const [query, setQuery] = useState<string>('');
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [error, setError] = useState<string | undefined>(undefined);
+  const [results, setResults] = useState<any>(undefined);
```

Comment:
```suggestion
  const [users, setUsers] = useState<User[] | undefined>(undefined);
```
Je recommande un meilleur naming, results est beaucoup trop générique 

Et Utiliser `any` est fortement déconseillé, on perd enormement de typesafety ! Ici par exemple, ligne 38 on va mapper sur les results alors qu'ils peuvent être undefined, ce qui peut causer une erreur au runtime !